npm.el

```diff
@@ -22,41 +22,47 @@
 
 ;;; Code:
 
-(require 'cl)
+(eval-when-compile
+  (require 'cl))
 (require 'compile)
+(require 'json)
 
 (defgroup npm nil
-  "Run npm in Emacs"
+  "Run npm in Emacs."
   :group 'tools)
 
 (defcustom npm-executable-path nil
-  "The path to the node and npm executables.
+  "The path to the `node' and `npm' executables.
 
-NIL if they should be looked up from the global path"
+If nil, npm.el shall look up from the global path."
   :type 'string
   :group 'npm)
 
-(setq +npm-dev-dir+ "~/dev")
-
-(setq npm-vars-name "hello-world")
-(setq npm-vars-desc "")
-(setq npm-vars-author (user-login-name))
-(setq npm-vars-git-user (user-login-name))
-(setq npm-vars-test-cmd "node test")
-(setq npm-vars-license "BSD")
-(setq npm-vars-main "index.js")
-(setq npm-vars-new-dependency "")
-(setq npm-vars-deps "")
-(setq npm-vars-keywords "")
-(setq npm-vars-last-search-keyword "")
-(setq npm-vars-version "0.0.0")
+(defcustom npm-dev-dir "~/dev"
+  "The path where npm will create new packages."
+  :type 'string
+  :group 'npm)
+
+(defvar npm-vars-name "hello-world")
```

Comment:
Is there an advantage to declaring these at the top level instead of just making them locals wherever they are used? since they will be one-offs per project they don't really seem like something a user will configure.